event-app-ui/src/components/EditProfile/EditProfile.jsx

```diff
@@ -0,0 +1,87 @@
+import "components/EditProfile/EditProfile.css";
+import PropTypes from "prop-types";
+import axios from "axios";
+import { useState } from "react";
+
+EditProfile.propTypes = {
+  setOpenProfile: PropTypes.func.isRequired,
+  fullname: PropTypes.string.isRequired,
+  email: PropTypes.string.isRequired,
+  tel: PropTypes.string.isRequired,
+  username: PropTypes.string.isRequired,
+};
+
+export default function EditProfile(props) {
+  const { setOpenProfile, fullname, email, tel, username } = props;
+  const [newFullname, setNewFullname] = useState(fullname);
+  const [newEmail, setNewEmail] = useState(email);
+  const [newTel, setNewTel] = useState(tel);
+  const [error, setError] = useState(false);
+
+  const editProfile = (e) => {
+    e.preventDefault();
+    if (
+      newFullname.length === 0 ||
+      newEmail.length === 0 ||
+      newTel.length === 0
+    ) {
+      setError(true);
+    } else {
+      axios.post(`http://localhost:3000/editProfile`, {
+        username: username,
+        fullname: newFullname,
+        email: newEmail,
+        tel: newTel,
+      });
+    }
+  };
+
+  return (
+    <div className="edit-profile">
+      <button className="close-button" onClick={() => setOpenProfile(false)}>
+        X
+      </button>
+      <input
+        className="edit-input"
+        type="text"
+        placeholder="fullname"
+        onChange={(e) => {
+          setNewFullname(e.target.value.trim());
+        }}
+      />
+      {error &&
+      (newFullname.length === 0 || newFullname.trim().length === 0) ? (
```

Comment:
You don't need to check the newFullname length is 0 if you're going to trim it anyway
```suggestion
      newFullname.trim().length === 0) ?
```